### PR TITLE
Features

### PR DESCRIPTION
**Pull request by CsiszerMihai**
Travel Packages Creation feature is pretty much done, looking forward to any feedback.

### CODE REVIEW

File: src/main/java/com/orange/team/controllers/TravelPackageController.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package com.orange.team.controllers;
```
Comment (AlexandruIliescu) [inline src/main/java/com/orange/team/controllers/TravelPackageController.java:1]: Please consider using the ticket name from Trello, for the PR title. `Features` doesn't reflect the specifics of the PR.

Also, please consider relevant messages for each commit, with some hints regarding the changes made in that commit.

image

File: src/main/java/com/orange/team/models/dtos/TravelPackageDTO.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package com.orange.team.models.dtos;
+
+import lombok.Data;
+import java.time.LocalDate;
+import java.time.Period;
+
+@Data
+
```
Comment (AlexandruIliescu) [inline src/main/java/com/orange/team/models/dtos/TravelPackageDTO.java:8]: Please consider to remove this unused empty row.

File: src/main/java/com/orange/team/models/dtos/TravelPackageDTO.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+public class TravelPackageDTO {
+
+    private long id;
+    private String name;
+    private String destination;
+    private String description;
+    private Period duration;
+    private double price;
+    private LocalDate availableDates;
```
Comment (AlexandruIliescu) [inline src/main/java/com/orange/team/models/dtos/TravelPackageDTO.java:17]: Do we need this as plural? We have only one date with `LocalDate`. If we need multiple dates, we should then consider a list?

File: src/main/java/com/orange/team/models/entities/TravelPackage.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import jakarta.persistence.*;
+import lombok.Data;
+
+import java.time.LocalDate;
+import java.time.Period;
+
+@Data
+@Entity
+@Table(name = "travel_packages")
+
```
Comment (AlexandruIliescu) [inline src/main/java/com/orange/team/models/entities/TravelPackage.java:12]: Please remove this empty row.

File: src/main/java/com/orange/team/models/entities/TravelPackage.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+public class TravelPackage {
```
Comment (AlexandruIliescu) [inline src/main/java/com/orange/team/models/entities/TravelPackage.java:13]: If `TravelPackageDTO` had an empty row after the class declaration, please be consistent and have the same practice in all classes.

File: src/main/java/com/orange/team/models/entities/TravelPackage.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(name = "package_name")
+    private String name;
```
Comment (AlexandruIliescu) [inline src/main/java/com/orange/team/models/entities/TravelPackage.java:18]: There is a reason to not have consistency between the entity's field and the name of the column? We can have both `packageName` & `package_name` or `name` and `name` (as a column). Given the fact that the DTO has this property `name`, I would use that.

File: src/main/java/com/orange/team/models/entities/TravelPackage.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    @Column(name = "destination")
+    private String destination;
+    @Column(name = "package_description")
+    private String description;
+    @Column(name = "duration")
+    private Period duration;
+    @Column(name = "price_per_person")
+    private double price;
+    @Column(name = "available_dates")
+    private LocalDate availableDates;
```
Comment (AlexandruIliescu) [inline src/main/java/com/orange/team/models/entities/TravelPackage.java:28]: If you change the name of this property on the DTO, please be aware to change it here as well.

File: src/main/java/com/orange/team/models/entities/TravelPackage.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    @Column(name = "package_name")
+    private String name;
+    @Column(name = "destination")
+    private String destination;
+    @Column(name = "package_description")
+    private String description;
+    @Column(name = "duration")
+    private Period duration;
+    @Column(name = "price_per_person")
+    private double price;
```
Comment (AlexandruIliescu) [inline src/main/java/com/orange/team/models/entities/TravelPackage.java:26]: Again, we should not have different names between the property and the column name.

File: src/main/java/com/orange/team/services/TravelPackageService.java [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+import com.orange.team.models.dtos.TravelPackageDTO;
+
+import java.util.List;
+
+public interface TravelPackageService {
+
+    TravelPackageDTO createTravelPackage(TravelPackageDTO travelPackageDTO);
+    List<TravelPackageDTO> getTravelPackages();
+    TravelPackageDTO updateTravelPackage(Long id, TravelPackageDTO travelPackageDTO);
+    void deleteTravelPackage(Long id);
```
Comment (AlexandruIliescu) [inline src/main/java/com/orange/team/services/TravelPackageService.java:12]: Can we maybe improve the name of this method? `deleteTravelPackageById` would be a better name?

File: src/main/java/com/orange/team/services/TravelPackageServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+        return travelPackages.stream()
+                .map(travelPackage -> objectMapper.convertValue(travelPackage, TravelPackageDTO.class))
+                .toList();
+    }
+
+    @Override
+    public TravelPackageDTO updateTravelPackage(Long id, TravelPackageDTO travelPackageDTO) {
+        TravelPackage existingPackage = travelPackageRepository.findById(id)
+        .orElseThrow(() -> new RuntimeException("Travel Package not found with id: " + id));
```
Comment (AlexandruIliescu) [inline src/main/java/com/orange/team/services/TravelPackageServiceImpl.java:47]: Please consider in the future to add here a custom exception. Given the fact that we update `TravelPackage` resource, we can have `UpdateTravelPackageException` designed and a global exception handler as a Controller Advice.

File: src/main/java/com/orange/team/services/TravelPackageServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+
+        existingPackage.setName(travelPackageDTO.getName());
+        existingPackage.setDestination(travelPackageDTO.getDestination());
+        existingPackage.setDescription(travelPackageDTO.getDescription());
+        existingPackage.setDuration(travelPackageDTO.getDuration());
+        existingPackage.setPrice(travelPackageDTO.getPrice());
+        existingPackage.setAvailableDates(travelPackageDTO.getAvailableDates());
```
Comment (AlexandruIliescu) [inline src/main/java/com/orange/team/services/TravelPackageServiceImpl.java:54]: Can we extract the logic between 49 - 54 into a distinct method? In this way, we can have a more easy to follow method.

File: src/main/java/com/orange/team/services/TravelPackageServiceImpl.java [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+        List<TravelPackage> travelPackages = travelPackageRepository.findAll();
+
+        return travelPackages.stream()
+                .map(travelPackage -> objectMapper.convertValue(travelPackage, TravelPackageDTO.class))
+                .toList();
+    }
+
+    @Override
+    public TravelPackageDTO updateTravelPackage(Long id, TravelPackageDTO travelPackageDTO) {
+        TravelPackage existingPackage = travelPackageRepository.findById(id)
```
Comment (AlexandruIliescu) [inline src/main/java/com/orange/team/services/TravelPackageServiceImpl.java:46]: What if `id` is `null`? Do we encounter an exception?

**Review by AlexandruIliescu — APPROVED** [review]

File: src/main/java/com/orange/team/controllers/TravelPackageController.java [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    @GetMapping("api/travel_packages")
+    public ResponseEntity<List<TravelPackageDTO>> getTravelPackages() {
+        return ResponseEntity.ok(travelPackageService.getTravelPackages());
+    }
+
+    @PutMapping("api/travel_packages")
+    public ResponseEntity<TravelPackageDTO> updateTravelPackage(Long id, TravelPackageDTO travelPackageDTO) {
+        return ResponseEntity.ok(travelPackageService.updateTravelPackage(id, travelPackageDTO));
```
Comment (antcsab) [inline src/main/java/com/orange/team/controllers/TravelPackageController.java:31]: You don't have updateTravelPackage, try to replace it with updateTravelPackagebyId

File: src/main/java/com/orange/team/controllers/TravelPackageController.java [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    @DeleteMapping("api/travel_packages")
+    public ResponseEntity<Void> deleteTravelPackage(Long id) {
+        travelPackageService.deleteTravelPackage(id);
```
Comment (antcsab) [inline src/main/java/com/orange/team/controllers/TravelPackageController.java:36]: You don't have deleteTravelPackage, try to replace it with deleteTravelPackagebyId

File: .gitignore
Comment (Buse-Ioan) [inline .gitignore:1]: 👍

**Review by Buse-Ioan — APPROVED** [review]

**Review by Buse-Ioan — APPROVED** [review]

**Review by Buse-Ioan — APPROVED** [review]

**Review by Buse-Ioan — APPROVED** [review]

**Review by Buse-Ioan — APPROVED** [review]

**Pull request merged**